proposed/SerializerGoals5.0.md

```diff
@@ -0,0 +1,281 @@
+# Overview and goals
+
+Apri 20th, 2020.
+
+This document provides options and recommendations for 5.0 to handle overlapping serializer goals including reach, faster cold startup performance, reducing memory usage, and smaller size-on-disk.
+
+## Reach
+Achieved by limiting or removing dependencies on Reflection.Emit and `Type.MakeGenericType`. This allows the serializer to be used on additional runtimes including [Xamarin iOS](https://github.com/dotnet/runtime/issues/31326).
+
+Ideally we add a high-performance emit replacement for targeted scenarios such as property setters, getters and constructors.
```

Comment:
This would address reach only if *everything* can be done with these.

Reply:
I re-worded this area to provide the full suite of options to consider since there are combination of refactoring, emit replacement, etc that can be used together.